code/src/components/TodoInput.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react"
+import { useDispatch } from "react-redux"
+import {
+  TextField,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Button,
+} from "@material-ui/core"
+import MomentUtils from "@date-io/moment"
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker,
+} from "@material-ui/pickers"
+import moment from "moment"
+
+import { tasks } from "reducers/tasks"
+
+export const TodoInput = (props) => {
+  const dispatch = useDispatch()
+  const [description, setDescription] = useState("")
+
+  const [selectedDate, setSelectedDate] = useState(Date.now())
+
+  const handleDateChange = (date) => {
+    setSelectedDate(date)
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    dispatch(
+      tasks.actions.addTask({
+        description,
+        date: moment(selectedDate).calendar({
+          sameDay: "[Today]",
+          nextDay: "[Tomorrow]",
+          nextWeek: "dddd",
+          lastDay: "[Yesterday]",
+          lastWeek: "[Last] dddd",
+          sameElse: "DD/MM/YYYY",
+        }),
+      })
+    )
+    props.onClose()
+    props.handleSnack()
+  }
+
+  const handleCancel = (e) => {
+    props.onClose()
+  }
+
+  return (
+    <>
+      <DialogTitle id="form-dialog-title">Add a new task</DialogTitle>
+      <DialogContent>
+        <TextField
+          type="text"
+          variant="standard"
+          name="name"
+          required
```

Comment:
This validation doesn't seem to work (it's possible to submit empty todos). Try adding a min value or add some validation to the `handleSubmit` function